tls/s2n_handshake.h

```diff
@@ -160,6 +160,15 @@ struct s2n_handshake {
 
     /* Set to 1 if the RSA verification failed */
     uint8_t rsa_failed;
+
+    /* Tracks if the client has seen a HelloRetryRequest */
+    unsigned client_received_hrr:1;
+
+    /* Tracks whether the server sent a HelloRetryRequest */
+    unsigned server_sent_hrr:1;
```

Comment:
it seems like server_sent_hrr is only used by the server, and maybe it could be added to the comments

Reply:
I've updated all the comments here to be more explicit about who uses the flag, and for what.